kratos/geometries/line_3d_2.h

```diff
@@ -685,6 +691,68 @@ class Line3D2 : public Geometry<TPointType>
         KRATOS_ERROR << "Jacobian is not square" << std::endl;
     }
 
+    /**
+     * @brief Returns whether given arbitrary point is inside the Geometry and the respective
+     * local point for the given global point
+     * @param rPoint The point to be checked if is inside o note in global coordinates
+     * @param rResult The local coordinates of the point
+     * @param Tolerance The  tolerance that will be considered to check if the point is inside or not
+     * @return True if the point is inside, false otherwise
+     */
+    bool IsInside(
+        const CoordinatesArrayType& rPoint,
+        CoordinatesArrayType& rResult,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+        ) override
+    {
+        PointLocalCoordinates( rResult, rPoint );
+
+        if ( std::abs( rResult[0] ) <= (1.0 + Tolerance) ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @brief Returns the local coordinates of a given arbitrary point
+     * @param rResult The vector containing the local coordinates of the point
+     * @param rPoint The point in global coordinates
+     * @return The vector containing the local coordinates of the point
+     */
+    CoordinatesArrayType& PointLocalCoordinates(
```

Comment:
@RiccardoRossi I rewrote the function to 3D and I  copy and simplified to the 2D case, as I said